### PR TITLE
Arachnio pins, Arduino IDE compile fix, USB import

### PR DESCRIPTION
**Pull request by logos-electromechanical**
This pull request contains three updates...

1) Implementation of Arachnio pin mapping. Self explanatory.

2) Import of USB modules from Arduino Due tree. These need to be modified to conform to local file naming conventions, and I have not tested them to see if the differences between the SAM3X and SAM4S require more changes.

3) Modified platform.txt so the code compiles as it is. This involved two changes. First, I added -I terms to the compiler.{c|cpp}.extra_flags variables that enable the IDE to find all the core files. Second, I replaced the compiler.libsam.c.flags variable in the compile recipes with compiler.arm.cmsis.path from the samd core. It still doesn't compile under the Arduino IDE, but the changes to make now are in the core files and will come under a different pull request.

### CODE REVIEW

File: variants/logos_arachnio_due/variant.cpp [hunk 1]
```diff
@@ -25,39 +25,72 @@ const Port Ports[NUM_PORTS]=
 {
   { .pGPIO=PIOA, .ulId=ID_PIOA },
   { .pGPIO=PIOB, .ulId=ID_PIOB },
-  { .pGPIO=PIOC, .ulId=ID_PIOC }
+  { .pGPIO=NULL, .ulId=NULL }
```
Comment (aethaniel) [inline variants/logos_arachnio_due/variant.cpp:28]: you can delete this line without problem

File: variants/logos_arachnio_due/variant.cpp [hunk 2]
```diff
@@ -25,39 +25,72 @@ const Port Ports[NUM_PORTS]=
...
+	{ PORTA, PIO_PA4A_TWCK0, GPIO_PERIPH_A, No_Analog_Channel, NOT_ON_PWM, NOT_ON_TIMER }, 		// D3/SCL
+	{ PORTB, PIO_PB3A_UTXD1, GPIO_PERIPH_A, ADC_Channel7, NOT_ON_PWM, NOT_ON_TIMER },			// D4
+	{ PORTB, PIO_PB4, GPIO_NOMUX, ADC_Channel7, NOT_ON_PWM, NOT_ON_TIMER },						// D5
+	{ PORTB, PIO_PB2A_URXD1, GPIO_PERIPH_A, ADC_Channel6, NOT_ON_PWM, NOT_ON_TIMER },			// D6
+	{ PORTB, PIO_PB4, GPIO_NOMUX, No_Analog_Channel, NOT_ON_PWM, NOT_ON_TIMER },				// D7
+	{ PORTB, PIO_PB7, GPIO_NOMUX, No_Analog_Channel, NOT_ON_PWM, NOT_ON_TIMER },				// D8
+	{ PORTB, PIO_PB5, GPIO_NOMUX, No_Analog_Channel, NOT_ON_PWM, NOT_ON_TIMER },				// D9
+	{ PORTB, PIO_PB2, GPIO_NOMUX, No_Analog_Channel, NOT_ON_PWM, NOT_ON_TIMER },				// D10
+	{ PORTB, PIO_PB1, GPIO_NOMUX, No_Analog_Channel, NOT_ON_PWM, NOT_ON_TIMER },				// D11
+	{ PORTB, PIO_PB11, GPIO_NOMUX, No_Analog_Channel, NOT_ON_PWM, NOT_ON_TIMER },				// D12
```
Comment (aethaniel) [inline variants/logos_arachnio_due/variant.cpp]: Isn't PB11 USB/DP?

**Comment by aethaniel** [comment]
@logos-electromechanical, can you remove the USB files from this PR, please? I would prefer to add them later with some refresh.

Thanks

File: variants/logos_arachnio_due/variant.cpp (hunk 2 above)
Comment (logos-electromechanical) [inline variants/logos_arachnio_due/variant.cpp]: Good catch... predictably, it should be PA11.

**Comment by aethaniel** [comment]
@logos-electromechanical, master has been updated with all my pending commits.
I will merge your PR after your update.

**Comment by logos-electromechanical** [comment]
Ok, I think I've got everything folded in.

**Pull request merged**